addons/payment_paypal/models/paypal_s2s.py

```diff
@@ -0,0 +1,231 @@
+# -*- coding: utf-'8' "-*-"
+
+import base64
+import json
+import logging
+import urllib
+import urllib2
+
+from odoo import api, models, _
+from odoo.tools.safe_eval import safe_eval
+
+_logger = logging.getLogger(__name__)
+
+
+class AcquirerPaypalS2s(models.Model):
+    _inherit = 'payment.acquirer'
+
+    # --------------------------------------------------
+    # SERVER2SERVER RELATED METHODS
+    # --------------------------------------------------
+
+    @api.model
+    def paypal_s2s_form_process(self, data):
+        values = {
+            'cc_holder_name': data.get('cc_holder_name'),
+            'cc_number': data.get('cc_number'),
+            'cc_expiry': data.get('cc_expiry'),
+            'cc_cvv': data.get('cc_cvv'),
+            'cc_brand': data.get('cc_brand'),
+            'acquirer_id': int(data.get('acquirer_id')),
+            'partner_id': int(data.get('partner_id'))
+        }
+        payment_method = self.env['payment.method'].sudo().create(values)
+        return payment_method.id
+
+    @api.multi
+    def _paypal_s2s_get_access_token(self):
+        path = '/v1/oauth2/token'
+        token_url = self._get_paypal_urls(self.environment)['paypal_rest_url'] + path
+        credentials = "%s:%s" % (self.paypal_client_id, self.paypal_client_secret)
+        encode_credential = base64.b64encode(
+            credentials.encode('utf-8')).decode('utf-8').replace('\n', '')
```

Comment:
Was there a problem with encoding? It seems to me that the client ID and secret are ascii characters anyway, so this should not be necessary.